pkg/workloads/cortex/client/cortex/client.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2019 Cortex Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+import os
+import types
+import subprocess
+import sys
+import shutil
+import yaml
+import urllib.parse
+
+import dill
+import requests
+from requests.exceptions import HTTPError
+
+
+class Client(object):
+    def __init__(
+        self, aws_access_key_id, aws_secret_access_key, operator_url, workspace="./workspace"
+    ):
+        self.operator_url = operator_url
+        self.workspace = workspace
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.headers = {
+            "CortexAPIVersion": "master",
+            "Authorization": "CortexAWS {}|{}".format(
+                self.aws_access_key_id, self.aws_secret_access_key
+            ),
+        }
+
+        pathlib.Path(self.workspace).mkdir(parents=True, exist_ok=True)
+
+    def deploy(self, deployment_name, api_name, model_path, pre_inference, post_inference):
+        working_dir = os.path.join(self.workspace, deployment_name)
+        api_working_dir = os.path.join(working_dir, api_name)
+        pathlib.Path(api_working_dir).mkdir(parents=True, exist_ok=True)
+
+        reqs = subprocess.check_output([sys.executable, "-m", "pip", "freeze"])
+
+        with open(os.path.join(api_working_dir, "requirements.txt"), "w") as f:
+            f.writelines(reqs.decode())
+
+        with open(os.path.join(api_working_dir, "request_handler.pickle"), "wb") as f:
+            dill.dump(
+                {"pre_inference": pre_inference, "post_inference": post_inference}, f, recurse=True
+            )
+
+        cortex_config = [
+            {"kind": "deployment", "name": deployment_name},
```

Comment:
Maybe it's ok for now, but would this not support deploying e.g. `iris/tensorflow` and then `iris/sklearn`? If we wanted to do that, would we keep track of a map of api name -> config so we could maintain the entire yaml? Let me know if you think we should discuss.